src/Symfony/Component/Scheduler/Command/DebugCommand.php

```diff
@@ -106,20 +107,30 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     /**
      * @return array{0:string,1:string,2:string}|null
      */
-    private static function renderRecurringMessage(RecurringMessage $recurringMessage, \DateTimeImmutable $date, bool $all): ?array
+    private static function renderRecurringMessage(RecurringMessage $recurringMessage, string $name, \DateTimeImmutable $date, bool $all): ?array
     {
-        $message = $recurringMessage->getMessage();
         $trigger = $recurringMessage->getTrigger();
 
+        $next = $trigger->getNextRunDate($date)?->format('r') ?? '-';
+        if ('-' === $next && !$all) {
+            return null;
+        }
+
+        $messages = $recurringMessage->getMessages(new MessageContext($name, $recurringMessage->getId(), $trigger, $date));
+        $message = null;
+        if (\is_array($messages) && 1 === \count($messages)) {
```

Comment:
What if this returns several messages ? It seems to ignore all except the first one, which looks like a confusing behavior for a debug command. It should represent the actual state of what will happen

Reply:
Tbh now that `getMessages()` may dynamically construct messages (e.g. using external sources for data), I think it shouldn't be called in this debug command. Perhaps we can replace it with `$recurringMessage->getProvider()::class` ?